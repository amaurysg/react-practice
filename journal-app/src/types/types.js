//here defined my types
//next configured in reducer this actions! 
const types = {
  login: '[Auth] Login',
  logout: '[Auth] Logout',

  uiSetError: '[UI] Set Error',
  uiRemoveError: '[UI] Remove Error',
  uiStartLoading: '[UI] Start loading',
  uiFinishLoading: '[UI] Finish loading',

}

export default types