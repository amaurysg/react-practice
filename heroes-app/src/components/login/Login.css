.login-container {
  background-image: url("https://i.pinimg.com/originals/e5/42/62/e5426248295aad18d6580667f3d43109.jpg");
  background-size: cover;
  width: 100vw;
  height: 100vh;
  left: calc(50% - 750px / 2 - 97px);
  top: calc(50% - 1334px / 2 - 299px);
  box-sizing: border-box;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
}
