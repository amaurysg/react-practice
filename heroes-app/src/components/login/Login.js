import React from "react";
import { useHistory } from "react-router";
import "./Login.css";

const Login = () => {
  //call hooks useHistory
  let history = useHistory();

  const handleClick = () => {
    // history.push("/");
    //When do click, so...
    //replace is better than push, but in this case.
    history.replace("/");
  };

  return (
    <div className="login-container">
      <div className="container">
        <h1>Logi</h1>
        <hr />
        <button className="btn btn-primary" onClick={handleClick}>
          Login
        </button>
      </div>
    </div>
  );
};

export default Login;
