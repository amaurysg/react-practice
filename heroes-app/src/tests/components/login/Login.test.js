import { mount } from "enzyme"
import { Router } from "react-router"
import AuthContext from "../../../auth/AuthContext"
import Login from "../../../components/login/Login"
import types from "../../../types/types"

describe('Test in Login', () => {

  const historyMock = {
    push: jest.fn(),
    replace: jest.fn(),
    location: {},
    listen: jest.fn(),
    createHref: jest.fn(),

  }

  const contextValues = {
    dispatch: jest.fn(),
    user: {
      logged: false
    }
  }


  const wrapper = mount(
    <AuthContext.Provider value={contextValues}>
      <Router history={historyMock}>
        <Login />
      </Router>
    </AuthContext.Provider>
  )

  //reset of mocks
  // afterEach(() => {
  //   jest.clearAllMocks()
  // })


  test('Debe hacer render correct', () => {

    expect(wrapper).toMatchSnapshot()


  })
  test('Debe de realizar el dispatch y navegacion', () => {

    const handleClick = wrapper.find('button').prop('onClick')
    handleClick()

    expect(contextValues.dispatch).toHaveBeenCalledWith({
      type: types.login,
      payload: {
        name: 'Amaury'
      }

    })

    expect(historyMock.replace).toHaveBeenCalledWith('/')
    localStorage.setItem('lastPathname', '/dc')
    // handleClick()
    // expect(historyMock.replace).toHaveBeenCalledWith('/dc')
    // expect(historyMock.replace).toHaveBeenCalledWith('/dc')


  })





})
