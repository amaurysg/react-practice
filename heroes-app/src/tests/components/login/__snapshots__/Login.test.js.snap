// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Login Debe hacer render correct 1`] = `
<div
  className="login-container"
>
  <div
    className="container"
  >
    <h1>
      Logi
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
